commit 9457a8be6e700f39e6b545f8db0edd30c0693700
Author: Ken Zhu <quic_guigenz@quicinc.com>
Date:   Tue Sep 6 11:11:20 2022 -0700

    nat46: fix sleeping warning back trace

    use spin_lock instead of mutex_lock since
    mutex_lock could sleep in the kernel packet process.

    Change-Id: I65c15a9f618ef296159884a0d6d742e66aaf6623
    Signed-off-by: Ken Zhu <quic_guigenz@quicinc.com>


[Andrei: rebased onto nat46 f2bb2eb; the spinlock switch is upstream (4020e96); only the rest remains.]
---
diff --git a/nat46/modules/nat46-glue.c b/nat46/modules/nat46-glue.c
index 5f06283..899c4de 100644
--- a/nat46/modules/nat46-glue.c
+++ b/nat46/modules/nat46-glue.c
@@ -47,7 +47,6 @@ nat46_instance_t *alloc_nat46_instance(int npairs, nat46_instance_t *old, int fr
   return nat46;
 }
 
-
 nat46_instance_t *get_nat46_instance(struct sk_buff *sk) {
   nat46_instance_t *nat46 = netdev_nat46_instance(sk->dev);
   spin_lock_bh(&ref_lock);
